piet/src/samples/mod.rs

```diff
@@ -76,53 +77,67 @@ struct Args {
 /// PNG to the path.
 ///
 /// The `prefix` argument is used for the file names of failure cases.
-pub fn samples_main(f: fn(usize, &Path) -> Result<(), BoxErr>, prefix: &str) -> Result<(), BoxErr> {
-    let args = match Args::from_env() {
-        Ok(args) => args,
-        Err(e) => {
+pub fn samples_main(f: fn(usize, &Path) -> Result<(), BoxErr>, prefix: &str) -> ! {
```

Comment:
So the compiler will only accept code for a function with return type `!` if all code paths diverge, i.e. there is no way the function can return. This has been stable since 1.0, the never type stuff is about promoting `!` to a full type. The reason we use it here rather than returning is we want to control error codes.

Reply:
The code would also compile if I put  `-> Result<(), BoxErr>`, or anthing else, since it will never be returned. `!` is the most correct way of communicating intent though, I think.

Reply:
If you look lower, all paths end with a call to `std::process::exit`, which is a diverging call.